### DISCUSSION TITLE
Minecraft-based version support policy

### DISCUSSION THREAD

**Discussion by AugustinMauroy** [Pumpkin-MC/Pumpkin#219] [Q&A] [answered]
We need to think about how to manage the pumpkin semver and the Minecraft semver. In addition, we need to see how to support the different versions of mc.

## Semver 

For me you just need a Minecraft major for a pumkin major. And a minor/patch equals a minor pumkin. And patches are exclusively for pumkin.

## Version support

For me, it would be necessary to have branches named `1.21.X` which is the version of MC. Then have a schedule that gives it Minecraft version support.

The problem with a potential diary is that it will be on one branch, so if it is updated it will be out of date on the other branches. That's why the idea of an org can be useful in this case. There would be an `admin` repository, another `core` repository and a `website` repository for the docs which is independent of the code itself.

**Comment by Snowiiii** [comment, marked as answer]
I don't want to have a bunch of branches for every Minecraft version. I would like to use tags when releasing a new Pumpkin version. 
Pumpkin will also not follow Minecraft's version scheme, We will release Pumpkin updates regardless of Minecraft updates, So the first version will be `1.0.0`.
There are no plans for having any kind of LTS version or maintain older Pumpkin versions which are compatible with older Minecraft versions

**Reply by Yimura** [reply]
My suggestion would be to tag the latest commit that works for a particular version of Minecraft. That way there are still some reference points to go back to.

Edit:
Tag only after a new version of Minecraft drops and Pumpkin is about to push a commit that breaks compatibility with that version.

**Comment by Snowiiii** [comment]
@AugustinMauroy the version policy was changed in https://github.com/Pumpkin-MC/Pumpkin/commit/be19110a241384eeecee4939834a97da977a1776

**Reply by AugustinMauroy** [reply]
Ohh wow super
